### PR TITLE
Don't run empty effects

### PR DESCRIPTION
**Pull request by hojberg**

### CODE REVIEW

File: src/effect.ts [hunk 1]
```diff
@@ -30,6 +30,10 @@ class _Effect {
     this.runner = Maybe(runner);
   }
 
+  isEmpty(): boolean {
+    return this.runner.map(() => false).getOrElse(true);
```
Comment (newyankeecodeshop) [inline src/effect.ts:34]: I would do `return this === EMPTY_EFFECT` or `return this.runner.isNothing()`.

File: src/program.tsx [hunk 2]
```diff
@@ -62,8 +62,10 @@ class Program extends React.Component<ProgramProps, ProgramState> {
   }
 
   _runEffect(effect: Effect): void {
-    console.log("[Gongfu] Running Effect");
-    effect.run(this._updater);
+    if (!effect.isEmpty()) {
+      console.log("[Gongfu] Running Effect");
+      effect.run(this._updater);
```
Comment (newyankeecodeshop) [inline src/program.tsx:67]: If an empty effect has a nothing instance for `this.runner`, shouldn't calling `effect.run()` be no less efficient?

Comment (newyankeecodeshop) [reply src/program.tsx:67]: I would ditch the console logging since the application should decide what and when to log?

Comment (hojberg) [reply src/program.tsx:67]: Hmm this is true actually. I think the logging was throwing me off as it was logging the empty effect *not* being run lol

Comment (hojberg) [reply src/program.tsx:67]: I'm going to move the logging to be debug enabled

**Pull request closed without merging**